ouroboros-network/src/Ouroboros/Network/TxSubmission/Outbound.hs

```diff
@@ -188,8 +174,8 @@ txSubmissionOutbound tracer maxUnacked TxSubmissionMempoolReader{..} _version co
 
           MempoolSnapshot{mempoolLookupTx} <- atomically mempoolGetSnapshot
 
-          let txidxs  = [ Map.lookup txid unackedMap | txid <- txids ]
-              txidxs' = catMaybes txidxs
+          let txidxs  = [ find (\(t,_) -> t == txid) unackedSeq | txid <- txids ]
```

Comment:
`unackedMap` has to be removed since it is keyed on `txid` and can't cope with duplicate txid in the same window.
The tx window size is 10. It is not configurable and requires a new protocol version to change.